C++/API/API.cpp

```diff
@@ -693,235 +633,218 @@ vector<string> iSENSE::get_dataset(string dataset_name, string field_name)
-  {
-    cerr << "\nError - please set a project ID!\n";
+  if (project_ID == EMPTY || project_ID.empty()) {
+    std::cerr << "\nError in method: post_json_key()\n";
+    std::cerr << "Please set a project ID!\n";
     return false;
   }
 
   // Check that a title and contributor key has been set.
-  if(title == "title" || title.empty())
-  {
-    cerr << "\nError - please set a project title!\n";
+  if (title == EMPTY || title.empty()) {
+    std::cerr << "\nError in method: post_json_key()\n";
+    std::cerr << "\nPlease set a project title!\n";
     return false;
   }
 
-  if(contributor_key.empty())
-  {
-    cerr << "\nErrror - please set a contributor key!\n";
+  if (contributor_key == EMPTY || contributor_key.empty()) {
+    std::cerr << "\nError in method: post_json_key()\n";
+    std::cerr << "\nPlease set a contributor key!\n";
     return false;
   }
 
   // If a label wasn't set, automatically set it to "cURL"
-  if(contributor_label == "label" || contributor_label.empty())
-  {
+  if (contributor_label == "label" || contributor_label.empty()) {
     contributor_label = "cURL";
   }
 
   // Make sure the map actually has stuff pushed to it.
-  if(map_data.empty())
-  {
-    cerr << "\nMap of keys/data is empty. You should push some data back to this object.\n";
+  if (map_data.empty()) {
+    std::cerr << "\nError in method: post_json_key()\n";
+    std::cerr << "Map of keys/data is empty.\n";
+    std::cerr << "Please push some data back to this object.\n";
     return false;
   }
 
-  // Should make sure each vector is not empty as well, since I had issues uploading
-  // if any ONE vector was empty. rSENSE complained about the nil class.
+  // Should make sure each vector is not empty as well, since I had issues
+  // uploading if any ONE vector was empty. rSENSE complained about a nil class.
 
   upload_URL = devURL + "/projects/" + project_ID + "/jsonDataUpload";
 
-  // Call the POST function, give it type 1 since this is a upload JSON by contributor key.
+  // Call the POST function, give it type 1
+  // since this is a upload JSON by contributor key.
   int http_code = post_data_function(POST_KEY);
 
   /*
   *  The iSENSE API gives us two response codes to check against:
   *  Success: 200 OK (iSENSE handled the request fine)
-  *  Failure: 401 Unauthorized (email / password or contributor key was not valid)
+  *  Failure: 401 Unauthorized (email/pw or contributor key was not valid)
   *  Failure: 422 Unprocessable Entity (email or contributor key was fine,
   *           but there was an issue with the request's formatting.
   *           Something in the formatting caused iSENSE to fail.)
   */
 
-  if(http_code == 200)
-  {
-    cout << "\n\nPOST request successfully sent off to iSENSE!\n";
-    cout << "HTTP Response Code was: " << http_code << endl;
-    cout << "The URL to your project is: " << dev_baseURL << "/projects/" << project_ID << endl;
+  if (http_code == 200) {

```

Comment:
May want to start making constants for the HTTP codes as well (200, 401, 404, 422).  We already have an HTTP code for `CURL_ERROR` so it'd be consistent to do the same for the rest of the codes.

200 = HTTP_SUCCESS
401 = HTTP_UNAUTHORIZED
404 = HTTP_NOT_FOUND
422 = HTTP_UNPROC_ENTITY